pkg/genericapiserver/server/serve_test.go

```diff
@@ -564,6 +566,312 @@ NextTest:
 	}
 }
 
+func TestBasicIpLimit(t *testing.T) {
+	var ipLimit ipBasedLimit
```

Comment:
I think this test would work well as a table. See https://dave.cheney.net/2013/06/09/writing-table-driven-tests-in-go